### PR TITLE
Upgrade Spotless plugin to 6.17.0

### LINKED ISSUE #82794: Spotless removes space before `instanceof` in pattern match
Spotless does this:

```
diff --git a/server/src/main/java/org/elasticsearch/cluster/coordination/Publication.java b/server/src/main/java/org/elasticsearch/cluster/coordination/Publication.java
index 3a48bb3f05d..506c0efe8ac 100644
--- a/server/src/main/java/org/elasticsearch/cluster/coordination/Publication.java
+++ b/server/src/main/java/org/elasticsearch/cluster/coordination/Publication.java
@@ -378,7 +378,7 @@ public abstract class Publication {

         private Exception getRootCause(Exception e) {
             if (e instanceof final TransportException transportException) {
-                if (transportException.getRootCause() instanceof final Exception rootCause) {
+                if (transportException.getRootCause()instanceof final Exception rootCause) {
                     return rootCause;
                 } else {
                     assert false : e;
```

An aesthetic thing for sure but I think it's better with the space than without.

### PR DESCRIPTION
**Pull request by pugnascotia**
Fixes #82794. Upgrade the Spotless plugin, which addresses the issue around formatting `instanceof` expressions. Formatting of statements including lambdas seems to have improved too.

### CODE REVIEW

**Comment by pugnascotia** [comment]
@elasticsearchmachine run elasticsearch-ci/bwc

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/PrevalidateNodeRemovalIT.java
Comment (mark-vieira) [inline server/src/internalClusterTest/java/org/elasticsearch/cluster/PrevalidateNodeRemovalIT.java:199]: I'm wondering if we can have it reformat these single-line lambdas as expressions instead of code blocks so it doesn't try to turn a single-line statement into a multi-line one. For example, this should just be:

```java
(handler, request, channel, task) -> logger.info("drop the check shards request")
```

**Review by mark-vieira — APPROVED** [review]
Thanks Rory. The updated lambda formatting is definitely way more compact, which I prefer, except in the single-line block situation which I think we should try and mitigate.

Also, should we add this to `.git-blame-ignore-revs` after merge?

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/PrevalidateNodeRemovalIT.java
Comment (pugnascotia) [reply server/src/internalClusterTest/java/org/elasticsearch/cluster/PrevalidateNodeRemovalIT.java:199]: The Eclipse formatter doesn't support these kinds of structural changes, it's just a formatter.

Spotless does now appear to support an additional tool called [cleanthat](https://github.com/diffplug/spotless/tree/main/plugin-gradle#cleanthat), which can do automated refactorings (see also the [homepage](https://github.com/solven-eu/cleanthat)). Unfortunately when I tried it, I got some weird dependency conflict error, and decided not to investigate further. I'll leave that to you 😁 I did get IntelliJ to look for all potential fix sites, and there's just over 1000. Also, making the change to remove lambda blocks doesn't always result in more readable code after a format. It needs more investigation 🤷 

I do indeed intend to update `.git-blame-ignore-revs` after merging.

**Comment by pugnascotia** [comment]
I updated `.git-blame-ignore-revs` in afcb0aab89f5176407424ef3c101ff9b115823e7.

**Comment by mark-vieira** [comment]
> Still, I'm pretty sure we could debate this stuff all day.

Indeed. I have the exact opposite preference and we'll likely never completely reach a consensus. The point is to just eliminate the debate by enforcing this and forcing people to deal with it 😉

**Pull request merged**
